plugins/Makefile.am

```diff
@@ -76,6 +76,10 @@ include experimental/stream_editor/Makefile.inc
 include experimental/ts_lua/Makefile.inc
 include experimental/url_sig/Makefile.inc
 
+if BUILD_URI_SIGNING_PLUGIN
```

Comment:
I think my preference would be of a more "generic" type of check. Check for the libraries that you need, and then build this plugin conditional on the existence of your requirements. That way, future additions are a lot more straight forward if they have the same or similar dependencies.

Reply:
That define is specified in `configure.ac:1283`:

    AM_CONDITIONAL([BUILD_URI_SIGNING_PLUGIN], [test ! -z "${LIBCJOSE}" -a ! -z "${LIBJANSSON}" -a "x${enable_pcre}" = "xyes" -a "x${has_libcrypto}" = "x1"])

I had to add tests for cjose, jansson, and crypto, but I was able to reuse the pcre test. If you have the required libraries, it will build the plugin. Since cjose and jansson are not usefully available via standard packagers, it will automatically statically link them if you compile them for it. (By default, everything will be dynamic. See the README for details.)

I modeled it after how the memcached plugin is handled. It's safe to say that autoconf is not my forte, though.

Reply:
Here's the memcached lines:

    #
    # Enable experimental/memcached_remap plugin
    #
    PKG_CHECK_MODULES([LIBMEMCACHED], [libmemcached >= 1.0], [have_libmemcached=yes], [have_libmemcached=no])
    AM_CONDITIONAL([BUILD_MEMCACHED_REMAP_PLUGIN], [test "x${have_libmemcached}" = "xyes"])

I talked with @PSUdaemon about it, and he suggested leaving the library detection in `configure.ac`, but then exporting all those variables and putting the logic on which libraries are required to build this in this `plugins/Makefile.am`. Is that what you're suggesting?

Reply:
If you did export "all those variables", what would that look like?